apps/main/src/loan/components/PageLoanCreate/Page.tsx

```diff
@@ -119,47 +120,44 @@ const Page = () => {
 
   useEffect(() => {
     if (isHydrated && curve) {
-      if (llamma) {
-        resetUserDetailsState(llamma)
-        fetchInitial(curve, isLeverage, llamma)
-        void fetchLoanDetails(curve, llamma)
+      if (market) {
+        resetUserDetailsState(market)
+        fetchInitial(curve, isLeverage, market)
+        void fetchLoanDetails(curve, market)
         setLoaded(true)
-      } else if (collateralDatasMapper) {
-        console.warn(
-          `Collateral ${rCollateralId} not found for chain ${rChainId}. Redirecting to market list.`,
```

Comment:
I made a choice here to remove the redirect, given that `api.getMintMarket` will throw an error if the market could not be found. This means the user now gets an error shown via the error boundary. I think explicitly failing like this is better than silently redirecting without telling the user what just happened. (And, I'm not sure how to 'catch' errors thrown in hooks 🙈 )

Reply:
This breaks the chain switcher, if you try to switch from any market page it will go to a non-existing market